WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/StateUtils.kt

```diff
@@ -0,0 +1,31 @@
+package org.wordpress.android.ui.prefs.categories.detail
+
+import org.wordpress.android.models.CategoryNode
+import org.wordpress.android.ui.prefs.categories.detail.SubmitButtonUiState.SubmitButtonDisabledUiState
+import org.wordpress.android.ui.utils.UiString
+
+data class UiState(
+    val categories: ArrayList<CategoryNode>,
+    val selectedParentCategoryPosition: Int,
+    val categoryName: String,
+    val submitButtonUiState: SubmitButtonUiState = SubmitButtonDisabledUiState
+)
+
+sealed class CategoryUpdateUiState {
```

Comment:
❓ Is there any reason why we haven't used a `Content/ Error/ Loading `single `UiState` pattern for a screen but split `CategoryUpdateUiState` from `UiState`?

Reply:
Sure, no worries. We can revisit it later. I haven't thought in depth about it, just wanted to know the direction we're taking.
[UI states: single stream or multiple streams?](https://developer.android.com/jetpack/guide/ui-layer#additional-considerations) guidelines should help us decide our path.